tests/test_spectrum_assignment.py

```diff
@@ -0,0 +1,378 @@
+                                        correct_disjn, compute_path_with_disjunction)
+
+TEST_DIR = Path(__file__).parent
+DATA_DIR = TEST_DIR / 'data'
+EQPT_FILENAME = DATA_DIR / 'eqpt_config.json'
+NETWORK_FILENAME = DATA_DIR / 'testTopology_auto_design_expected.json'
+SERVICE_FILENAME = DATA_DIR / 'testTopology_services_expected.json'
+
+@pytest.fixture()
+def eqpt():
+    """ common setup for tests: builds network, equipment and oms only once
+    """
+    equipment = load_equipment(EQPT_FILENAME)
+    return equipment
+
+@pytest.fixture()
+def setup(eqpt):
+    """ common setup for tests: builds network, equipment and oms only once
+    """
+    equipment = eqpt
+    # fix band to be independant of changes in json file
+    equipment['SI']['default'].f_min = 191300000000000.0
+    equipment['SI']['default'].f_max = 196100000000000.0
+    network = load_network(NETWORK_FILENAME, equipment)
+    p_db = equipment['SI']['default'].power_dbm
+    p_total_db = p_db + lin2db(automatic_nch(equipment['SI']['default'].f_min,\
+        equipment['SI']['default'].f_max, equipment['SI']['default'].spacing))
+    build_network(network, equipment, p_db, p_total_db)
+    oms_list = build_oms_list(network, equipment)
+    return network, oms_list
+
+def test_oms(setup):
+    """ tests that the oms is between two roadms, that there is no roadm or transceivers in the oms
+        except end points, checks that the id of oms is present in the element and that the element
+        oms id is consistant
+    """
+    network, oms_list = setup
+    for oms in oms_list:
+        assert isinstance(oms.el_list[0], Roadm) and isinstance(oms.el_list[-1], Roadm)
+        for i, elem in enumerate(oms.el_list[1:-2]):
+            assert not isinstance(elem, Roadm)
+            assert elem  in network.nodes()
+            assert elem.oms.oms_id == oms.oms_id
+            print(f'expected {elem.uid}, obtained {oms.el_id_list[i+1]}')
+            assert elem.uid == oms.el_id_list[i+1]
+
+@pytest.mark.parametrize('nmin', [-288, -260, -300])
+@pytest.mark.parametrize('nmax', [480, 320, 450])
+def test_aligned(nmin, nmax, setup):
+    """ checks that the oms grid is correctly aligned. Note that bitmap index uses guardband
+        on both ends so that if nmin, nmax = -200, +200, min/max navalue in bitmap are
+        -224, +223, which makes 223 -(-224) +1 frequencies
+    """
+    network, oms_list = setup
+    # f_min = 193.1e12 - 280 * 0.00625e12
+    # f_max = 193.1e12 + 320 * 0.00625e12
+    # f_min = 1931.1 f_max = 195.1
+    grid = 0.00625e12
+    guardband = 0.15e12
+    nguard = 24
+    freq_min = 193.1e12 + nmin * 0.00625e12
+    freq_max = 193.1e12 + nmax * 0.00625e12
+    print('initial spectrum')
+    print(nvalue_to_frequency(oms_list[10].spectrum_bitmap.freq_index_min) * 1e-12,
+          nvalue_to_frequency(oms_list[10].spectrum_bitmap.freq_index_max) * 1e-12)
+    # checks initial values consistancy
+    ind_max = len(oms_list[10].spectrum_bitmap.bitmap) - 1
```

Comment:
I don't know what `10` means. It's used at a lot of places below, so let's give it a name so that a reader can keep up, e.g.:

```python
last_fiber = oms_list[10]
# ...
assert nvalmin == last_fiber.spectrum_bitmap.freq_index_min
```

Reply:
OK Done